src/main/java/io/smilo/api/peer/PeerReceiver.java

```diff
@@ -92,12 +92,24 @@ public void getNewData() {
     public void broadcastNewBlockRequest() {
         if(networkState.getCachupMode()) {
             try {
-                //Sleep for a bit, wait for responses before requesting more data.
-                Thread.sleep(300);
+
                 //Broadcast request for new block(s)
-                Long blockNum =  blockStore.getLatestBlockHeight() + 1;
-                LOGGER.info("Requesting block " + blockNum + "...");
-                peerClient.broadcast("GET_BLOCK " + blockNum);
+                Long blockNum =  blockStore.getLatestBlockHeight();
+                Long blockGoal = networkState.getTopBlock();
+                int max_blocks = 25;
+
+                if (blockGoal - blockNum < 25) {
+                    max_blocks = (int) (blockGoal - blockNum);
+                }
+
+                for (int i = 1; i < max_blocks; ++i) {
```

Comment:
Is this logic correct? Say max_blocks is 5 then the iterated values would be:
1 ( < 5 == true)
2 ( < 5 == true)
3 ( < 5 == true)
4 ( < 5 == true)
5 ( < 5 == false)

Which means the iterated block count is max_blocks - 1. 

Fix could be:
`for(int i = 0; i <= max_blocks; ++i)`